embedded-graphics/src/primitives/line.rs

```diff
@@ -65,76 +60,318 @@ impl Dimensions for Line {
 impl Line {
     /// Create a new line
     pub const fn new(start: Point, end: Point) -> Self {
-        Line { start, end }
+        Self { start, end }
+    }
+}
+
+/// Which side of the center line to draw on
+///
+/// Imagine standing on `start`, looking ahead to where `end` is. `Left` is to your left, `Right` to
+/// your right.
+#[derive(Copy, Clone, Eq, PartialEq, Ord, PartialOrd, Hash, Debug)]
+pub enum Side {
+    /// DOCS LOL
+    Left,
+    /// DOCS LOL
+    Right,
+}
+
+impl Side {
+    fn swap(&mut self) {
+        match self {
+            Self::Left => *self = Self::Right,
+            Self::Right => *self = Self::Left,
+        }
+    }
+}
+
+/// Current state of each parallel line drawn
+#[derive(Copy, Clone, Eq, PartialEq, Ord, PartialOrd, Hash, Debug)]
+struct ParallelLineState {
+    start: Point,
```

Comment:
`start` is a misleading name for this field, because it's the current point and not the start of the line.

Reply:
Renamed to `current_point`.